R/plotting.R

```diff
@@ -219,26 +231,37 @@ names(.all_labs_values) <- .all_aes_names
 
   # Dispatch to different plotting commands, depending on whether X/Y are groupable.
   if (group_X && group_Y) {
-    plot_cmds <- .griddotplot(..., color_set=color_set)
+    plot_cmds <- .griddotplot(
+      ..., color_set=color_set,
+      color_map=color_map, color_name=color_name, color_discrete=group_color
+    )
 
   } else if (group_X && !group_Y) {
     cmds$todo[["group"]] <- "plot.data$GroupBy <- plot.data$X;"
     fill_set <- (color_set && group_color)
     if (fill_set) {
       cmds$todo[["fill"]] <- "plot.data$FillBy <- plot.data$ColorBy"
     }
-    plot_cmds <- .violin_plot(..., color_set=color_set, fill_set=fill_set)
+    plot_cmds <- .violin_plot(
+      ..., color_set=color_set, fill_set=fill_set,
+      color_map=color_map, color_name=color_name, color_discrete=group_color
```

Comment:
If `colouring` is NULL, `group_color` could conceivably be undefined. In general, there are too many `color_` variables being passed around here; I would prefer to see the `.process_colorby_choice` moved into `.create_plot`, and the entire thing being passed to the various plotting functions (given that the choice of color is the same).

Reply:
Agreed that i felt bad about adding yet another `color_*` argument to the function call. However, when I prefer to separate code refactoring from new features.
In other words, I'd prefer to merge this PR first, and open another branch for the refactoring.

Reply:
I will think about this - the choices of where to put things are quite complex...

Reply:
This can be done with some functional programming magic, where `.process_colorby_choice` also defines a function  it is passed to `.create_plot`. All of the necessary information is embedded in the function closure, thus avoiding the need to pass a chain of `color_*` variables while preserving the separation of responsibilities between functions.